File: FeedbackHub.Domain/Dto/Feedback/FeedbackRevisionDto.cs
namespace FeedbackHub.Domain.Dto.Feedback
{
    public class FeedbackRevisionDto
    {
        public int Id { get; set; }
        public int FeedbackId { get; set; }
        public string ChangedBy { get; set; }
        public DateTime ChangedAt { get; set; }
        public List<FeedbackChangedFieldDto> ChangedFields { get; set; } = new();
    }

    public class FeedbackChangedFieldDto
    {
        public string FieldName { get; set; }
        public string OldValue { get; set; }
        public string NewValue { get; set; }
        public string DisplayName { get; set; }
    }

}

File: feedbackhub.client/src/types/feedback/FeedbackRevisionDto.ts
export interface FeedbackChangedFieldDto {
    FieldName: string;
    OldValue: string;
    NewValue: string;
    DisplayName: string; 
}

export interface FeedbackRevisionDto {
    Id: number;
    FeedbackId: number;
    ChangedBy: string;
    ChangedAt: string;
    ChangedFields: FeedbackChangedFieldDto[];
}
